client/zanata-client-commands/src/main/java/org/zanata/client/commands/PutUserCommand.java

```diff
@@ -21,22 +22,45 @@ public PutUserCommand(PutUserOptions opts) {
     }
 
     public void run() throws Exception {
-        Account account = new Account();
-        account.setEmail(getOpts().getUserEmail());
-        account.setName(getOpts().getUserName());
-        account.setUsername(getOpts().getUserUsername());
-        account.setPasswordHash(getOpts().getUserPasswordHash());
-        account.setApiKey(getOpts().getUserKey());
+
+        String userUsername = getOpts().getUserUsername();
+        String passwordHash = getOpts().getUserPasswordHash();
+        String userEmail = getOpts().getUserEmail();
+
+        Account account = getClientFactory().getAccountClient().get(userUsername);
+
+        if (account == null) {
+            log.info("Creating new account {}", userUsername);
+            account = new Account();
+            if (userEmail == null || userUsername == null) {
+                throw new RuntimeException("New user's name and email must be specified");
+            } else if (StringUtils.isBlank(passwordHash)) {
+                log.warn("Unusable passwordHash set, user will need to use the Forgot Password feature");
+                account.setPasswordHash("NULL");
```

Comment:
@seanf I've tried this now, and I still get the error. I suspect it has something to do with the actual construction of the REST request body, but I don't understand the code well enough to pinpoint where.